src/wallet/lightning.ts

```diff
@@ -134,6 +135,38 @@ export class LightningWallet {
       return { rHash: rHash.toString("hex"), paymentRequest };
     }
   }
+
+  public async assertLndDetails(
+    selfPublicKey: string,
+    chain: string,
+    network: string
+  ): Promise<void> {
+    const getinfo = await this.lnd.lnrpc.getInfo();
```

Comment:
We could store that once during the initialization and then just reuse it for every action. Otherwise you will make that call all the time again :man_shrugging:

Reply:
Yes I went through the same thought process but then decided to not make assumptions or early optimisation hence the lack of caching.